code/common/utils.q

```diff
@@ -0,0 +1,87 @@
+\d .checkinputs
+
+//- utils for reading in config
+readtableproperties:{[tablepropertiepath]
+  pfield:$[()~.Q.pt;`;.Q.pf];
+  .lg.o[`readtableproperties;"loading table properties"];
+  table:`tablename`proctype xkey readcsv[tablepropertiepath;"ssssstsss"];                                                            //read in table from file
+  alltable:?[table;enlist(in;`proctype;enlist`all`);0b;()];                                                                          //find any instance of the use "all" or blank for proctype
+  table:table,![alltable;();0b;(enlist`proctype)!enlist(enlist `hdb)],![alltable;();0b;(enlist`proctype)!enlist(enlist `rdb)];       //join rdb and hdb entries for any "all" or blank entries
+  table:![table;enlist(in;`proctype;enlist`all`);0b;`symbol$()];                                                                     //remove "all" or blank entries from table
+  table:?[table;$[.proc.proctype=`gateway;();enlist(=;`proctype;`.proc.proctype)];0b;()];
+  table:update pfield^partitionfield,.eodtime.datatimezone ^ datatimezone, .eodtime.rolltimeoffset ^ rolltimeoffset,.eodtime.rolltimezone^rolltimezone from table;
+  .lg.o[`readtableproperties;"Table properties successfully loaded"];
+  :table;
+      };
+
+readcheckinputs:{[checkinputspath] spliltcolumns[readcsv[checkinputspath;"sbs*"];`invalidpairs;`]};
+
+readcsv:{[path;types]
+  if[not pathexists path:hsym path;'path];
+  :(types;1#",")0:path;
+ };
+
+pathexists:{[path] path~key path};
+
+spliltcolumns:{[x;columns;types]@[x;columns;spliltandcast;types]};
+
+spliltandcast:{[x;typ]typ$"|"vs/:x};
+
+
+//- functions:
+//- (i) .dataaccess.getmetainfo - mapping from tablename to metainfo;
+
+getmetainfo:{
+  .lg.o[`getmetainfo;"Loading meta info"];
+  tlist:exec tablename from .checkinputs.tablepropertiesconfig;
+  partfield:$[()~key`.Q.pf;`;.Q.pf];
+  metainfo:1!/:`columns`types`attributes xcol/:`c`t`a#/:0!/:meta each tlist;
+  :1!flip(`tablename`partfield`metas`proctype)!(tlist;partfield;metainfo;.proc.proctype);
+  .lg.o[`getmetainfo;"metainfo successfully loaded"]
+ };
+
+//- misc utils
+getvalidparams:{[]checkinputsconfig`parameter};
+getrequiredparams:{[]exec parameter from checkinputsconfig where required};
```

Comment:
aren't these defined in `code/common/checkinputs.q`? do they need to be defined here as well? (as far as I can see with a quick search they aren't referenced elsewhere in this file)

Reply:
There was code lifted from `checkinputs.q` and moved to `utils.q` . I'll remove the duplicate code.